examples/prom/maxwell_global_rom.cpp

```diff
@@ -0,0 +1,555 @@
+    a->AddDomainIntegrator(new VectorFEMassIntegrator(*sigma));
+
+    // 15. Assemble the parallel bilinear form and the corresponding linear
+    //     system, applying any necessary transformations such as: parallel
+    //     assembly, eliminating boundary conditions, applying conforming
+    //     constraints for non-conforming AMR, static condensation, etc.
+    if (static_cond) {
+        a->EnableStaticCondensation();
+    }
+    a->Assemble();
+
+    OperatorPtr A;
+    Vector B, X;
+    a->FormLinearSystem(ess_tdof_list, x, *b, A, X, B);
+    assembleTimer.Stop();
+
+    // 16. Offline Phase: Solve the system AX=B using PCG with an AMS preconditioner.
+    if(fom || offline)
+    {
+        // 17. Solve the full order linear system A X = B
+        if (pa)
+        {
+#ifdef MFEM_USE_AMGX
+            MatrixFreeAMS ams(*a, *A, *fespace, muinv, sigma, NULL, ess_bdr, useAmgX);
+#else
+            MatrixFreeAMS ams(*a, *A, *fespace, muinv, sigma, NULL, ess_bdr);
+#endif
+            CGSolver cg(MPI_COMM_WORLD);
+            cg.SetRelTol(1e-12);
+            cg.SetMaxIter(1000);
+            cg.SetPrintLevel(1);
+            cg.SetOperator(*A);
+            cg.SetPreconditioner(ams);
+            solveTimer.Start();
+            cg.Mult(B, X);
+            solveTimer.Stop();
+        }
+        else
+        {
+            if (myid == 0)
+            {
+                cout << "Size of linear system: "
+                     << A.As<HypreParMatrix>()->GetGlobalNumRows() << endl;
+            }
+
+            ParFiniteElementSpace *prec_fespace =
+                (a->StaticCondensationIsEnabled() ? a->SCParFESpace() : fespace);
+            HypreAMS ams(*A.As<HypreParMatrix>(), prec_fespace);
+            HyprePCG pcg(*A.As<HypreParMatrix>());
+            pcg.SetTol(1e-12);
+            pcg.SetMaxIter(500);
+            pcg.SetPrintLevel(2);
+            pcg.SetPreconditioner(ams);
+            solveTimer.Start();
+            pcg.Mult(B, X);
+            solveTimer.Stop();
+        }
+        // 18. take and write snapshot for ROM
+        if (offline)
+        {
+            bool addSample = generator->takeSample(X.GetData(), 0.0, 0.01);
+            generator->writeSnapshot();
+            delete generator;
+            delete options;
+        }
+    }
+
+    // 19. The online phase
+    if (online) {
+        // 20. read the reduced basis
+        assembleTimer.Start();
+        CAROM::BasisReader reader(basisName);
+        spatialbasis = reader.getSpatialBasis(0.0);
+        numRowRB = spatialbasis->numRows();
+        numColumnRB = spatialbasis->numColumns();
+        if (myid == 0) printf("spatial basis dimension is %d x %d\n", numRowRB,
+                                  numColumnRB);
+
+        // libROM stores the matrix row-wise, so wrapping as a DenseMatrix in MFEM means it is transposed.
+        DenseMatrix *reducedBasisT = new DenseMatrix(spatialbasis->getData(),
+                numColumnRB, numRowRB);
```

Comment:
This seems unnecessary. The same object is defined in the global ROM for Poisson equation, both in C++ and Python side. Can @andersonw1 @dreamer2368 take care of it? Thanks!

Reply:
I deleted this line and everything runs with the same result, so it seems this part was unnecessary.

Reply:
Thanks. I'll take care of the Poisson equation.